c2/external/external.go

```diff
@@ -0,0 +1,87 @@
+package external
+
+import (
+	"github.com/vulncheck-oss/go-exploit/c2/channel"
+	"github.com/vulncheck-oss/go-exploit/output"
+)
+
+type Server struct {
+	Read    any
+	Write   any
+	flags   func()
+	init    func()
+	run     func(int)
+	meta    func(*channel.Channel)
+	Channel *channel.Channel
+}
+
+type External interface {
+	Configure(*Server)
+	SetFlags(func())
+	SetInit(func())
+	SetRun(func(int))
```

Comment:
We should also probably set a ok return value on this one too, probably others as well.